File: app/schema/charwidth.py
"""Schema for character width analysis."""

from typing import List, Optional
from pydantic import BaseModel


class CharWidthAnalysis(BaseModel):
    """Character width analysis result."""
    text: str
    total_chars: int
    fullwidth_chars: List[str]
    halfwidth_chars: List[str]
    other_chars: List[str]
    fullwidth_count: int
    halfwidth_count: int
    other_count: int
    is_all_fullwidth: bool
    is_all_halfwidth: bool
    has_mixed_widths: bool


class CharWidthResponse(BaseModel):
    """Response for character width analysis."""
    original_query: str
    is_chahan_related: bool
    analysis: CharWidthAnalysis
    converted_fullwidth: str
    converted_halfwidth: str
    message: str

File: app/frontend/src/types/charwidth.ts
export interface CharWidthAnalysis {
  text: string;
  total_chars: number;
  fullwidth_chars: string[];
  halfwidth_chars: string[];
  other_chars: string[];
  fullwidth_count: number;
  halfwidth_count: number;
  other_count: number;
  is_all_fullwidth: boolean;
  is_all_halfwidth: boolean;
  has_mixed_widths: boolean;
}

export interface CharWidthResponse {
  original_query: string;
  is_chahan_related: boolean;
  analysis: CharWidthAnalysis;
  converted_fullwidth: string;
  converted_halfwidth: string;
  message: string;
}